--- ASTWriteStmtNode.java.orig	2007-07-31 09:19:09.000000000 -0600
+++ ASTWriteStmtNode.java	2007-07-31 09:19:34.000000000 -0600
@@ -1,2 +1,12 @@
+/*******************************************************************************
+ * Copyright (c) 2007 University of Illinois at Urbana-Champaign and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     UIUC - Initial API and implementation
+ *******************************************************************************/
 package org.eclipse.photran.internal.core.parser;
 
