diff -uNr org.eclipse.ptp.proxy/include/proxy_attr.h org.eclipse.ptp.proxy_slurm/include/proxy_attr.h
--- org.eclipse.ptp.proxy/include/proxy_attr.h	2009-04-14 15:08:29.000000000 +0800
+++ org.eclipse.ptp.proxy_slurm/include/proxy_attr.h	2009-04-14 15:12:40.000000000 +0800
@@ -56,10 +56,12 @@
  * Node attributes
  */
 #define NODE_STATE_ATTR				"nodeState"
+#ifndef HAVE_SLURM_SLURM_H
 #define 	NODE_STATE_UP				"UP"
 #define 	NODE_STATE_DOWN				"DOWN"
 #define 	NODE_STATE_ERROR			"ERROR"
 #define 	NODE_STATE_UNKNOWN			"UNKNOWN"
+#endif
 #define NODE_NUMBER_ATTR			"nodeNumber"
 #ifdef HAVE_SYS_BPROC_H
 #define NODE_GROUP_ATTR				"nodeGroup"
